bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -2767,15 +2767,31 @@ void ReplicaImp::onViewsChangeTimer(Timers::Handle timer)  // TODO(GG): review/u
 
     if (!hasPendingRequest) return;
 
-    const uint64_t diffMilli1 = duration_cast<milliseconds>(currTime - timeOfLastStateSynch).count();
-    const uint64_t diffMilli2 = duration_cast<milliseconds>(currTime - timeOfLastViewEntrance).count();
-    const uint64_t diffMilli3 = duration_cast<milliseconds>(currTime - timeOfEarliestPendingRequest).count();
-
-    if ((diffMilli1 > viewChangeTimeout) && (diffMilli2 > viewChangeTimeout) && (diffMilli3 > viewChangeTimeout)) {
-      LOG_INFO(
-          VC_LOG,
-          "Ask to leave view=" << curView << " (" << diffMilli3 << " ms after the earliest pending client request).");
-
+    const uint64_t timeFromLastStateTransfer = duration_cast<milliseconds>(currTime - timeOfLastStateSynch).count();
+    const uint64_t timeFromLastViewEntrance = duration_cast<milliseconds>(currTime - timeOfLastViewEntrance).count();
+    const uint64_t timeFromEarliestPendingRequest =
+        duration_cast<milliseconds>(currTime - timeOfEarliestPendingRequest).count();
+    const uint64_t timeFromLastExecution = duration_cast<milliseconds>(currTime - timeOfLastExecution).count();
+
```

Comment:
Just a proposal, we might log if we have a Client operation beyond half of the VC timeout, but less than the VC timeout just as an early indication that the Replica is getting behind.